File: 560-SubarraySumEqualsK/560-SubarraySumEqualsK.py
# Last updated: 07/04/2026, 11:38:33
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        c= 0 
        curr = 0 
        hashmap = {}



        for i in range(len(nums)) :
            curr += nums[i]

            if curr == k :
                c+=1

            if curr-k in hashmap:
                c+=hashmap[curr-k]

            if curr in hashmap:
                hashmap[curr] +=1
            else:
                hashmap[curr] =1 



        return c

File: 560-SubarraySumEqualsK/560-SubarraySumEqualsK.java
// Last updated: 15/04/2026, 12:20:06
1class Solution {
2    public int subarraySum(int[] nums, int k) {
3        HashMap<Integer , Integer> resdue = new HashMap<>();
4        resdue.put( 0 , 1);
5        int ans = 0 ;
6        int cur = 0 ;
7
8        for ( int i : nums){
9            cur+=i;
10
11            int res = cur-k;
12            ans +=(resdue.getOrDefault(res, 0));
13            resdue.put(cur,resdue.getOrDefault(cur,0)+1);
14
15        }
16        return ans;
17    }
18}